### PR TITLE
Add linux disk IO metrics to system/process

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message
Part of https://github.com/elastic/beats/issues/7461

This actually pulls in the changes from elastic-agent-system-metrics, and adds linux-only disk IO metrics to system/process.

No actual code changes here, see https://github.com/elastic/elastic-agent-system-metrics/pull/114

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
- [x] I have made corresponding changes to the documentation
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -62308,6 +62308,75 @@ format: bytes
...
+The number of bytes written to the storage layer.
+
+type: long
+
+--
+
+*`system.process.io.read_char`*::
++
+--
+The number of bytes read fromo read(2) and similar syscalls.
```
Comment (belimawr) [inline metricbeat/docs/fields.asciidoc:62347]: Typo
```suggestion
The number of bytes read from read(2) and similar syscalls.
```

File: metricbeat/docs/fields.asciidoc [hunk 2]
```diff
@@ -62308,6 +62308,75 @@ format: bytes
 
 --
 
+[float]
+=== io
+
+Disk I/O Metrics, as forwarded from /proc/[PID]/IO. Available on Linux only.
```
Comment (belimawr) [inline metricbeat/docs/fields.asciidoc:62314]: I was looking on my fs and the whole path is lowercase (e.g: `/proc/2847/io`), I believe the docs should reflect it as well.
```suggestion
Disk I/O Metrics, as forwarded from /proc/[PID]/io. Available on Linux only.
```

File: metricbeat/module/system/process/_meta/fields.yml [hunk 3]
```diff
@@ -111,6 +111,31 @@
           format: bytes
           description: >
             The shared memory the process uses.
+    - name: io
+      type: group
+      description: Disk I/O Metrics, as forwarded from /proc/[PID]/IO. Available on Linux only.
```
Comment (belimawr) [inline metricbeat/module/system/process/_meta/fields.yml:116]: Same here
```suggestion
      description: Disk I/O Metrics, as forwarded from /proc/[PID]/io. Available on Linux only.
```

File: metricbeat/module/system/process/_meta/fields.yml [hunk 4]
```diff
@@ -111,6 +111,31 @@
...
+          description: The number of bytes this process cancelled, or caused not to be written.
+        - name: read_bytes
+          type: long
+          description: The number of bytes fetched from the storage layer.
+        - name: write_bytes
+          type: long
+          description: The number of bytes written to the storage layer.
+        - name: read_char
+          type: long
+          description: The number of bytes read fromo read(2) and similar syscalls.
```
Comment (belimawr) [inline metricbeat/module/system/process/_meta/fields.yml:129]: Typo
```suggestion
          description: The number of bytes read from read(2) and similar syscalls.
```

**Review by belimawr — CHANGES_REQUESTED** [review]

**Review by belimawr — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]

**Pull request merged**